Merge changes from the original repo

arconamagi opened the pull request:
<!--
Thank you for your pull request. Please make sure this PR is scoped to one change, and that any added or changed code includes tests with greater than 80% code coverage. See https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#testing-prebidjs for documentation on testing Prebid.js.
-->

## Type of change
<!-- Remove items that don't apply and/or select an item by changing [ ] to [x] -->
- [ ] Bugfix
- [ ] Feature
- [ ] New bidder adapter  <!--  IMPORTANT: if checking here, also submit your bidder params documentation here https://github.com/prebid/prebid.github.io/tree/master/dev-docs/bidders --> 
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [x] Other

## Description of change
<!-- Describe the change proposed in this pull request -->
This is just a merge of changes from the original repo.

xlozinguez commented on modules/yieldmoBidAdapter.js:
```diff
@@ -94,12 +94,15 @@ var YieldmoAdapter = function YieldmoAdapter() {
   // expose the callback to the global object:
   $$PREBID_GLOBAL$$.YMCB = function(ymResponses) {
-    if (ymResponses) {
+    if (ymResponses && ymResponses.length > 0) {
```
@cdoher01 this will work if the response is an empty array, however the reviewer indicated that the response was an empty object (https://github.com/prebid/Prebid.js/pull/1415)... let's make sure the backend returns an array in every case.

xlozinguez replied:
or maybe add a check on type to confirm it's an array as well

onaydenov committed:
Fidelity adapter: TCFv2 support, kubient alias. (#5302)

* TCFv2 support, kubient alias

* TCFv2 support, kubient alias

TheMediaGrid committed:
Update sync url for grid and gridNM Bid Adapters (#5304)

* Added TheMediaGridNM Bid Adapter

* Updated required params for TheMediaGridNM Bid Adapter

* Update TheMediGridNM Bid Adapter

* Fix tests for TheMediaGridNM Bid Adapter

* Fixes after review for TheMediaGridNM Bid Adapter

* Add support of multi-format in TheMediaGrid Bid Adapter

* Update sync url for grid and gridNM Bid Adapters

estavillo committed:
Gumgum add in video (#5284)

* add in-video product line

* add in-video product line

* add unit tests and fix dependencies.

Co-authored-by: Estavillo <fernando@gumgum.com>

montu1996 committed:
DailyhuntBid Adapter: Add video support with Refactor/Optimizing (#5226)

* dailyhunt bidder refactor

* refactor dailyhunt bid adapter support

* native prebid server support

* video support

* native support

* win notice url support

* fix undefined object access issues

* dh bidder small bug fixes

* change endpoint

* added basic gulp test for dailyhunt

* adding test case and support for outstream

* body2 support in native request

* create md

* test mode support

* chnage endpoint to prod and remove console log

* change md

* adding accept-encoding:gzip

* chnage eslint to default

* remove array.prototype.find

* fix review changes

Co-authored-by: Mitesh Thakor <mitesh.thakor@dailyhunt.in>

GeneGenie committed:
Adtelligent new features (#5203)

* Adtelligent support adpods

* Adtelligent support bid chunks

* Adtelligent support userId, schain

* Adtelligent Rename params to be supported in post

* Coppa support

* Rewritten tests

* Add param transform for aid for ServerAdapter

* Lint

vladi-mmg committed:
New adapter - videofy (#5259)

* Change publisherId to zoneId
Add gdpr
Add supply chain
Add video media type

* Remove comments

* Fix unit test coverage

* fix request id bug
add vastXml to video response

* Remove bid response default sizes

* Change endpoint url

* Add unit test for vastXml

* Change end point

* Remove trailing-space

* Add onBidWon function

* New adapter - videofy

robertrmartinez committed:
fallback to defaultGdprScope if cmp undefined (#5291)

omerDotan committed:
RTD bug fix (#5087)

* real time data module,
browsi sub module for real time data,
new hook bidsBackCallback,
fix for config unsubscribe

* change timeout&primary ad server only to auctionDelay
update docs

* support multiple providers

* change promise to callbacks
configure submodule on submodules.json

* bug fixes

* use Prebid ajax

* tests fix

* browsi real time data provider improvements

* real time data module,
browsi sub module for real time data,
new hook bidsBackCallback,
fix for config unsubscribe

* change timeout&primary ad server only to auctionDelay
update docs

* support multiple providers

* change promise to callbacks
configure submodule on submodules.json

* bug fixes

* use Prebid ajax

* tests fix

* browsi real time data provider improvements

* RTD bug fixes

* #5087 review fixes

* RTD fixes

* use core-js lib find
init googletag if needed

robertrmartinez committed:
adding check if gdpr module ran enforcment logic (#5178)

* adding check if gdpr module ran enforcment logic

* adding tests for new hasValidated flag

Co-authored-by: rmartinez <Rachael24!>

robertrmartinez committed:
Floors new signals (#5295)

* Price Floors skipRate debug by query string

Rubicon Analytics log floors skipRate

* new floor signal fetchFailed

* change fetchFailed to fetchStatus

* rubi analytics looks for fetchStatus

sumit116 committed:
change iabSubCatId name to primaryCatId (#5303)

* change iabSubCatId name to primaryCatId

* change in adapter

Co-authored-by: sumit sharma <sumit.sharma1@SYNPUNLT9114TFL.appnexus.com>

bretg committed:
Prebid 3.21.0 Release

bretg committed:
Increment pre version

sumit116 committed:
Add publisher_id to appnexus adapter (#5244)

* Add publisher_id to appnexus adapter

* update transformBidParams fn

Co-authored-by: sumit sharma <sumit.sharma1@SYNPUNLT9114TFL.appnexus.com>

vyatsun committed:
Add Onomagic Bidder Adapter (#5155)

preved-medved committed:
AdPartner bidadapter init (#5196)

* AdPartner bidadapter init

* Change endpoint path and domain

* Add mock banner for test launch

* Add mock banner for test launch

K-JBoon committed:
Blue Billywig adapter - Add site config to request similarly to app config (#5273)

* add Blue Billywig adapter

* Blue Billywig Adapter - update according to review feedback

* Blue Billywig Adapter - update to try and pass CircleCI

* Remove the last for .. of in bluebillywigBidAdapter.js, hopefully...

* Update bluebillywigBidAdapter test parameters to match renderer to rendererCode rename

* Blue Billywig - Also pass through site config with OpenRTB request

Co-authored-by: Klaas-Jan Boon <klaas-janboon@live.nl>

MelodyLi2017 committed:
remove uri-encoding (#5307)

Co-authored-by: Melody Li <melody@yieldmo.com>

vedantseta committed:
Media.net improvements (#5282)

Co-authored-by: vedant.s <vedant.s@media.net>

Fawke committed:
adWMGAnalytics - refactor unit tests (#5310)

mk0x9 committed:
VIS.X: remove bidderCode from bidResponse (#5313)

adam-browning committed:
OneVideo: Excluded DAP validation for Multi-Format adUnits support (#5309)

* Excluded DAP validation if publisher is using Multi-Format adUnits

* Update package-lock.json

* refractoring of if statement to check bid.mediaTypes.video just once

* removed redundent statement

Fawke committed:
Improvements for SizeMappingV2 (advanced size mapping) module (#5283)

* basic implementatino done

* fix indentical ad unit count bug

* modules/sizeMappingV2.js

* revert change

* add missing log message

* fix existing failing test cases

* add additional tests for function getAdUnitDetail particularly for cacheHits and instance check

* add instructive comments and rename file to advanced_size_mapping

* flatten ternary operator call

Commit:
add onBidWon function, add bidder adapter version to bid requests (#5263)

GeneGenie committed:
Adtarget bid adapter (#5296)

* Adtarget frist release

* Fix

* Fix lint

* Coppa support

* Added unit tests

* revert package

* Added unit tests

susyt committed:
removes referencing of digitrust library (#5316)

Commit:
ucfunnel adapter update request parameter (#5278)

* Add a new ucfunnel Adapter and test page

* Add a new ucfunnel Adapter and test page

* 1. Use prebid lib in the repo to keep updated
2. Replace var with let
3. Put JSON.parse(JSON.stringify()) into try catch block

* utils.getTopWindowLocation is a function

* Change to modules from adapters

* Migrate to module design

* [Dev Fix] Remove width and height which can be got from ad unit id

* Update ucfunnelBidAdapter to fit into new spec

* Correct the endpoint. Fix the error of query string

* Add test case for ucfunnelBidAdapter

* Fix lint error

* Update version number

* Combine all checks on bid request

* Add GDPR support for ucfunnel adapter

* Add in-stream video and native support for ucfunnel adapter

* Remove demo page. Add more test cases.

* Change request method from POST to GET

* Remove unnecessary comment

* Support vastXml and vastUrl for video request

* update TTL to 30 mins

* Avoid using arrow function which is not discuraged in mocha

* ucfunnel tdid support

* ucfunnel fix error message in debug mode

* ucfunnel adapter add bidfloor parameter

* ucfunnel adapter support CCPA

* ucfunnel adapter native support clicktrackers

* ucfunnel adapter change cookie sync setting

* ucfunnel adapter update request parameter

Co-authored-by: root <root@ubuntu.members.linode.com>
Co-authored-by: Ryan Chou <ryanchou0210@gmail.com>
Co-authored-by: jack.hsieh <moonnight8520@gmail.com>

adxpremium committed:
AdxPremium Analytics - bug fixes, handling timeout, etc. (#5182)

* AdxPremium Analytics - bug fixes, handling timeout, etc.

* AdxPremium Analytics - bug fixes, handling timeout, etc.

* AdxPremium Analytics adapter update

* AdxPremium Analytics update

* unit testing added

* adxpremiumAnalyticsAdapter_spec.js

onlsol committed:
Update Dspx adapter (#5277)

* Add gdpr_consent support for Dspx Adapter

* Add video support for dspx adapter

* Update Video Test Parameters for Dspx Adapter

* fix referrer in DSPX adapter

* Fix Description in DSPX adapter

Co-authored-by: Alexander <avj83@list.ru>

Fawke committed:
remove only keyword (#5324)

pm-harshad-mane committed:
PubMatic analytics documentation (#5322)

* added support for pubcommon, digitrust, id5id

* added support for IdentityLink

* changed the source for id5

* added unit test cases

* changed source param for identityLink

* simple example

* added limitations

* added contact details

* example text

harpere committed:
Secure creative update (#5285)

* replaced manually set secureCreative adServerDomain with automatic postMessage event.source.orgin

* using event.origin rather than event.source.origin in secureCreative postMessage()

* updated x-domain iframe render example

Co-authored-by: Eric Harper <eharper@rubiconproject.com>

jsnellbaker committed:
update webdriverio and node 12 support (#5236)

* update webdriverio and node 12 support

* clean-up some commented code

* updates to supported node version and other minor updates and wording

* fix lint error

* add description files for fixtures

jsnellbaker committed:
Prebid 3.22.0 release

jsnellbaker committed:
increment pre version

thuyhq committed:
Update Usersync Url For Quantumdex bid adapter (#5327)

We have upgraded the new user cookie storage system. Where everything is gathered in one place

Tosh39 committed:
fix wipesAdapter CPM culclation (#5330)

susyt committed:
GumGum: handling misconfigured requests (#5212)

* uses encodeURIComponent inline

* adds test for jcsi param

* adds request delay depending on previous response

* adds inVideo param

susyt committed:
GumGum: update jcsi object values (#5258)

* updates jcsi object

* adds test for jcsi

* fix lint

* updates due to prebid rollback

tim-hm committed:
* Add Glimpes Protocol Bidding Adaptor (#5293)

* Initial submission of adaptor

* Currently only supports Banner

* 100% code coverage from tests

* Author Tim Holmes-Mitra <tim@glimpseprotocol.io>

jsalis committed:
fix secure protocol check for beachfront video (#5318)

Co-authored-by: John Salis <johnsalis@beachfrontmedia.com>

matthieularere-msq committed:
new bidder adapter: mediasquareBidAdapter (#5317)

* add adapter mediasquare

* Update mediasquareBidAdapter.md

* Update mediasquareBidAdapter.js

* test-coverage

* Update mediasquareBidAdapter.js

* Update mediasquareBidAdapter_spec.js

* Update mediasquareBidAdapter_spec.js

* Update mediasquareBidAdapter_spec.js

* Object.values unsupported by IE11

konduit-dev committed:
Konduit Accelerate module with 'send all bids' support (#5247)

* Adding Konduit module

* Removed superfluous arguments passed to obtainVastUrl function

* Removed superfluous arguments passed to obtainVastUrl function.

* Build trigger (empty commit)

* Module documentation updated according to the comments

* Logic in obtainVastUrl function updated according to the review comment.

* Removed hook, enabled eslint

* Merged recent prebid changes

* New method is introduced to process a bid and return dynamic CPM data

* New Konduit Analytics adapter responsible for client auction stats collection

* Updated konduit analytics adapter .md file

* Fixed linter issue with more than 1 blank line used

* Use '$prebid.version$' instead of the $$PREBID_GLOBAL$$.version

* Updated unit tests

* Enable "Send all bids" support

* Updated konduitWrapper.md file

* Updated links in konduitWrapper.md

* Updated spec file (unit tests)

* Added Konduit Prebid module version

Co-authored-by: Max Shevchenko <max.shevchenko@sigma.software>
Co-authored-by: Alexander Kislitsyn <alexander.kislitsyn@sigmaukraine.com>

icflournoy committed:
parrableIdSystem: Read legacy ID and optout cookies and migrate to new cookie storage implementation (#5219)

* Add unit coverage for parrableIdSystem getId callback

* PBID-14: Pass uspString to Parrable as us_privacy query parameter

* PBID-14: Simplify parrableIdSystem us_privacy test

* PBID-14: Only send us_privacy to Parrable when a value exists

* PBID-11: Read new Parrable compound cookie _parrable_id

Migrating from legacy _parrable_eid cookie. The new cookie contains ibaOptout and ccpaOptout status fields

* Remove path check from parrableIdSystem url test

* PBID-11: Integrate Parrable compound cookie, consolidating old cookies

* PBID-11: Update parrableIdSystem requestBids hook test to support compound cookie value

* PBID-11: Small refactor to parrableIdSystem spec to support compound cookie

* PBID-11: Handle legacy ibaOptout as truthy value when migrating to compound cookie

* PBID-11: Add parrableIdSystem spec tests covering migration of legacy cookies

* PBID-11: Remove storage documentation from test pages and userId module docs

* PBID-11: Remove SUBMODULES_THAT_ALWAYS_REFRESH_ID feature from userId system

* PBID-11: Use better serialize implementation for Parrable compound cookie

* PBID-11: Update parrableIdSystem interface documentation

* Add missing extension to mock xhr import

* PBID-11: Try to access eid property only when parrableId object exists

* PBID-11: Construct parrableId from legacy cookies in same manner as compound cookie

* Use hardcoded expiration date for legacy cookies

* parrableIdSystem: Relocate new unit test from upstream

* PBID-39: Fallback to cookie values when backend response is missing components

Also handle another missed callback scenario if the response object parses to nothing

nailyk committed:
Teads adapter: Support deal targeting (#5270)

susyt committed:
GumGum: adds new param (#5297)

* adds in new videoPubID param

* adds test

scritchley committed:
Avocet bid adapter (#5262)

* adds avocet bid adapter

* modules/avocetBidAdapter.js: fixes lint issues

* adds id5 support and test/code improvements

Commit:
Event updates (#5288)

* event updates

* removed char

* add comment regarding removal of hb_wiurl and hb_bidid

* optimized tests

* fix for str validation

* add test helper method to reset wurl map

* update resetWurlMap

* update test description

* optimized calls to setConfig in tests

* add timestamp to videoCache cached bid

* revert unintended change to adapterManager.js

* update imports ordering for cleaner diff

* update string validation to use isStr

* fix rename event to plural form

* update event to events in tests

mmuras-gemius committed:
AdOcean adapter - support for sizes defined in Prebid configuration. (#5337)

* AdOcean adapter - support for multiple sizes

* AdOcean adapter - tests - use normal functions instead of arrow ones for consistency

* AdOcean adapter - support for multiple sizes - changed way of sending dimensions

* AdOcean adapter - change separator between sizes group

* AdOcean adapter - small fix in buildRequest

uditalias committed:
Vidazoo: Feature support usp consent (#5111)

* feat(module): multi size request

* fix getUserSyncs
added tests

* update(module): package-lock.json from master

* feat(module): add usp consent support

* added uspConsent test

Co-authored-by: roman <shmoop207@gmail.com>

jsalis committed:
add optional video param for tagid (#5344)

Co-authored-by: John Salis <johnsalis@beachfrontmedia.com>

h12media committed:
Add module H12 Media (#5328)

* Add module H12 Media

* Add module H12 Media

* Add module H12 Media — fix gdpr info

* Add module H12 Media — fix find

redaguermas committed:
No bid version 1.2.6 (#5323)

* Enable supplyChain support

* Added support for COPPA

* - Added support for Safeframe creative.
- Added support for out-stream video.

* Changed the protocol from JSON to string because Exceptions cause a slowdown on the browser.

* Change in package.json

* Fix error

* Changed startsWidth() to indexOf()

* Fix lint error. Not sure why my gulp lint does not detect this error.

* Fix indexOf not available on IE 11

Co-authored-by: Reda Guermas <reda.guermas@nobid.io>

osazos committed:
AdagioBidAdapter 2.2.2 (#5347)

* Adagio: params auto-detection

Improve user integration with auto-detection for adUnitElementId and environment params.

* Adagio: update config example in .md file

braizhas committed:
added CCPA support, gvlid to adform and adformOpenRTB adapters (#5214)

raz-adroll committed:
Native support for NextRoll adapter (#5319)

* Add native support

* Add response testing

* DRY test

* Change required from bool to int

* Set mediaType

* Fixes objects

* Fixes object access

* Remove ad property, only set it for banner

* Update tests

* Moving hardcoding values to constants

* Update docs with native information

skazedo committed:
Adding back ZEDO adapter with changes to make it compatible to latest prebid (#5276)

* initial commit

* updated contact and tag details

* changes ti support the renderers

* changes to pass dimId

* fixed names of internal mapping

* added comment

* added gdpr param to request and other fixes

* modified api url

* fix

* fixed the secure api call

* rolled back video event callback till we support it

* updated doc with video details

* added bid won and timeout pixel

* added testcase for bid events

* modified testcase

* fixed the url logged

* tag param values passed ot renderer

* added a conditioal check

* changes to support new param to adserver for purpose of tracking

* passed param to renderer

* missing variable defined

* changes to pass schain

* fix

* added protocol to url

* fixed test for protocol

* changed urls to secure only

* fixes to make it compatible to head

* added support for ccpa

* Fixed path of find module

* Remove package.json

* updated test example with new setup

* Adding back the file

* trying to resolve conflict

Co-authored-by: Sanoska Gonsalves <sgonsalves@zedo.com>

uditalias committed:
Vidazoo Adapter: Feature/direct deal targeting (#5343)

* feat(module): multi size request

* fix getUserSyncs
added tests

* update(module): package-lock.json from master

* feat(client): send deal id with each request

Co-authored-by: roman <shmoop207@gmail.com>

itaynave committed:
Add support for aliases (#5342)

* Support new aniview bid adapter

* Support new aniview bid adapter

* Support new aniview bid adapter

* Support new aniview bid adapter

* Fix Consent parameters

* Update aniviewBidAdapter.js

V3 support

* Update aniviewBidAdapter.js

* Update aniviewBidAdapter.js

Update refererInfo

* Update aniviewBidAdapter.js

Fix tabs and spaces

* Update aniviewBidAdapter.js

fixes

* Update aniviewBidAdapter.js

* Update aniviewBidAdapter.js

Add ccpa support

* Update aniviewBidAdapter.js

Typo

* Update aniviewBidAdapter.js

* Update aniviewBidAdapter.js

* Fix size and sample

Fixed sizes from playerSize
Updated md sample

* Fix tabs

* Fix sizes

* Recheck

* Add tgt parameter

* Update sample

* Add support for cookie sync + tests

* Add support for cookie sync + tests

* Add support for cookie sync + tests

* Support aliases

Support aliases

* Update

Update

* Fix lint

Fix lint

* Update spec

Update spec

goosemanjack committed:
Fix to issue #5141 exception thrown in pbjs.requestBids when DigiTrus… (#5333)

* Fix to issue #5141 exception thrown in pbjs.requestBids when DigiTrust does not init.
Safe extract value of id or null.

* Code syntax change to overcome linter issue.

pm-harshad-mane committed:
PubMatic analytics adapter to support bidCpmAdustment values (#5354)

* added support for pubcommon, digitrust, id5id

* added support for IdentityLink

* changed the source for id5

* added unit test cases

* changed source param for identityLink

* set gross ecpm using originalCpm

* using bidGrossCpmUSD to set eg

* lint

* fixed old cases

* added some tests

* removed comments

* removed commented code , and unused import

* en related changes in test cases

* using pbjs.getHighestCpmBids, added test cases around it

* passing highestCpmBids

* fixed test cases

* added notes

* using getGlobal() instead of $$PREBID_GLOBAL$$

Commit:
Prebid 3.23.0 Release

Commit:
increment prebid version

lowendavid committed:
improveStartDelay after documentation update default value is now 1 and 0 does not exist. (#5361)

bretg committed:
Updating docs header info fields (#5366)

Related to https://github.com/prebid/prebid.github.io/pull/2056

npeceniak committed:
Add placement_type and position parameters to spotxBidAdapter (#5364)

* Add min_duration and max_duration parameter to spotxBidAdapter

* Add placement_type and position parameters to spotxBidAdapter

Co-authored-by: Nick Peceniak <npeceniak@spotx.tv>

ckbo3hrk committed:
Configurable user-sync types support (#5359)

robertrmartinez committed:
Delaying removal of floor data for 3 seconds (#5360)

Commit:
PBS Bid Adapter: allow setting site params (#4973)

* add site config value to oRTB request

* update to copy site.page and site.publisher.id if not defined in config site object

tsturtew committed:
Support for ID5 (#5345)

* Add support for bidderRequest.refererInfo in Adhese Adapter.

* Add support for bidderRequest.refererInfo in Adhese Adapter.

* Jira AD-2462 / add tlall is consent string is present

* Added 'adhese' attribute to bid that contains meta data - Jira AD-2642

* added DALE to adhese determination

* extra config option: no format, but use size array as format string

* Added more values to originData

* Revert "Added more values to originData"

This reverts commit 57f003318cfb7dc2f9d1a97015a7639823b181a6.

* Pulled Sander's changes & added more data to originData

* Adhese bid adapter - final version

* add origin and originInstance

make ad.origin and ad.originInstance

* add id5 id to request as x5

* Add support for bidderRequest.refererInfo in Adhese Adapter.

* added DALE to adhese determination

* updated tests to include new fields

* Added test for ID5 id

Co-authored-by: Mateusz Michalowski <mateusz@adhese.eu>
Co-authored-by: Mateusz Michalowski <mateusz@michalowski.biz>
Co-authored-by: Tim Sturtewagen <tim@adhese.com>
Co-authored-by: Kim Van Crombrugge <kim@adhese.eu>
Co-authored-by: Sander <sander@adhese.eu>
Co-authored-by: westerschmal <30859973+westerschmal@users.noreply.github.com>

TheMediaGrid committed:
Added keywords parameter to TheMediaGrid Bid Adapter (#5353)

* Added TheMediaGridNM Bid Adapter

* Updated required params for TheMediaGridNM Bid Adapter

* Update TheMediGridNM Bid Adapter

* Fix tests for TheMediaGridNM Bid Adapter

* Fixes after review for TheMediaGridNM Bid Adapter

* Add support of multi-format in TheMediaGrid Bid Adapter

* Update sync url for grid and gridNM Bid Adapters

* TheMediaGrid Bid Adapter: added keywords adUnit parameter

* Update TheMediaGrid Bid Adapter to support keywords from config

thuyhq committed:
Apply some updates to Valueimpression bid adapter (#5356)

- Update new adapter endpoint & usersync url
- Use window.top instead window object
- Use bidRequest.referrerInfo object to get page url, referrer
- Tracking ad position by size called targetKey

susyt committed:
GumGum: removes deprecated banner sizes default (#5372)

* uses encodeURIComponent inline

* adds test for jcsi param

* adds request delay depending on previous response

* adds inVideo param

* removes deprecated bidRequest.sizes

SKOCHERI committed:
UserId SharedId submodule (#5315)

* Adding sharedid submodule

* Updating with Shared ID Module

* SharedID test and sharedid eids

* Shared ID md changes

* Shared ID md changes

* Shared ID changes

* Apply suggestions from code review

Co-Authored-By: Brad Rodriguez <brodrigu@gmail.com>

* Applying review suggestions

* Apply suggestions from code review

Co-Authored-By: Brad Rodriguez <brodrigu@gmail.com>

* Reformatting and reorganizing sharedId submodule

* Reformatting and reorganizing sharedId submodule

* Shared Id generation changes

* Adding cookie Sync

* Decode and sync cookie

* Updating endpoint

* Updaitng eids.md

* Configured sync

* Refactor and md update

* Refactoring

* Refactoring

* Updating sync to seconds

* Updating configuration

* Reformatting

* Reformatting

* Reformatting

* Fixing review comments

* Changes to id value

* Updating documentation

* Documentation update

* Resolving merge conflicts

* updating userid_example.html

* Fixing review comments on test to separate sharedid opt out tests

* Moving sharedID generation within sharedId module

* Moving sharedID generation within sharedId module

Co-authored-by: skocheri <skocheri@rubiconproject.com>
Co-authored-by: Brad Rodriguez <brodrigu@gmail.com>

xwang202 committed:
Freewheel - Converted the ComponentId property to be 'prebid' in request (#5320)

* freewheel-ssp fix issue on playerSize of bidRequest

* freewheel add dealId property in bidResponse

* freewheel convert ComponentID to 'prebid' in request

* freewheel add new parameter 'componentSubId' in request

vladi-mmg committed:
Marsmedia & videofy adapters - Add onTimeout & onSetTargeting (#5352)

* Change publisherId to zoneId
Add gdpr
Add supply chain
Add video media type

* Remove comments

* Fix unit test coverage

* fix request id bug
add vastXml to video response

* Remove bid response default sizes

* Change endpoint url

* Add unit test for vastXml

* Change end point

* Remove trailing-space

* Add onBidWon function

* New adapter - videofy

* Marsmedia & Videofy - Add onTimeout onSetTargeting

* Create sendbeacon function

pm-harshad-mane committed:
PubMatic bid adapter to support price floors module (#5387)

* added support for pubcommon, digitrust, id5id

* added support for IdentityLink

* changed the source for id5

* added unit test cases

* changed source param for identityLink

* added support for floors module

* using floorModule to set floor

* removed commented console.log statements

colbertk committed:
Triplelift Adaptor: Use Floors Module (#5329)

* access floor from floors module

* int -> float and null check

* let -> const

jsnellbaker committed:
appnexusBidAdapter - fix video params (#5394)

* appnexusBidAdapter - fix video params

* remove mimes field

pm-harshad-mane committed:
PubMatic adds support for bidUserIdAsEids (#5397)

* added support for pubcommon, digitrust, id5id

* added support for IdentityLink

* changed the source for id5

* added unit test cases

* changed source param for identityLink

* id value should be string only

* id value should be string; changed test data

* pubmatic using userIdAsEids

commented unnecessary code
PubMatic will no longer read Digitrust from config or independent library setup
PubMatic will no loner read TTDID from config or independent library setup

* chnages in test cases for userIdAsEids

commented unnecessary test cases
PubMatic will no longer read Digitrust from config or independent library setup
PubMatic will no loner read TTDID from config or independent library setup

* deleted commented code and test cases

* deleted a commented line

* lint effect

lowendavid committed:
External ids (#5351)

* externalIds (SIM-657) Added external ids and associated unit test.

* externalIds - Added DEFAULT_PARAMS_WITH_EIDS that contains the external id information for the unit test

* externalIds (SIM-657) uses createEidsArray found in the eid.js

* externalIds cleanup spaces and tabs, adding a missing import for createEids in eids.js

* externalIds adding a space where it is needed to fix a wonderful test.

curlyblueeagle committed:
33Across: CCPA Compliance + Schain support (#5365)

* check gdpr in buildRequest

* User sync based on whether gdpr applies or not

* check if consent data exists during user sync

* split user sync into further branches: 1) when gdpr does not apply 2) when consent data is unavailable

* contribute viewability to ttxRequest

* update tests

* remove window mock from tests

* use local variables

* introduce ServerRequestBuilder

* add withOptions() method to ServerRequestBuilder

* add semicolons

* sync up package-lock.json with upstream/master

* stub window.top in tests

* introduce getTopWindowSize() for test purpose

* reformat code

* add withSite() method to TtxRequestBuilder

add withSite() method to TtxRequestBuilder

* add isIframe() and _isViewabilityMeasurable()

* handle NON_MEASURABLE viewability in nested iframes

* consider page visibility, stub utils functions getWindowTop() and getWindowSelf()

* contribute viewability as 0 for inactive tab

* add prebidjs version to ttx request

* send caller as an array

* fix JSDoc in utils.js

* send viewability as non measurable when unable to locate target HTMLElement, add warning message

* introduce mapAdSlotPathToElementId()

* introduce getAdSlotHTMLElement(), add logging

* introduce mapAdSlotPathToElementId()

* update logging in ad unit path to element id mapping

* rephrase logging, fix tests

* update adapter documentation

* remove excessive logging

* improve logging

* revert change

* fix return of _mapAdUnitPathToElementId()

* improve logging of _mapAdUnitPathToElementId()

* do not use Array.find()

* return id once element is found

* return id once element is found

* let -> const

* Removing killswitch behavior for GDPR

* Updated comments to reflect current gdpr logic

* URI encode consent string

* Updated example site ID to help Prebid team e2e test our adapter

* send page url in ortb

* Removed redundant pageUrl default

* Restored package-log.json that mirrors prebid's repo

* Sending USP string during buildRequest

* Adding USP consent data to user sync

* add unit test for syncing without bidrequest

* Changed to uspConsent to make the connatation consistent

* Resetting adapter state in adapter after user sync rather than exposing it.

* removed console log

* Adding schain info

* remove setting empty format ext

* better tests invalid values

* removing validation of schain

* Fixed lint errors

Co-authored-by: Gleb Glushtsov <gleb.glushtsov@33across.com>
Co-authored-by: Gleb Glushtsov <glebglushtsov@users.noreply.github.com>
Co-authored-by: Gleb Glushtsov <gleb.glushtsov@gmail.com>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-29-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@admins-MacBook-Pro.local>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-137-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-96-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@AHEGDE-MAC.local>
Co-authored-by: Aparna Hegde <ahegde@AHEGDE-MAC.fios-router.home>

mshuhaliia committed:
added waardex adapter (#5349)

* added waardex adapter

* removed support http from adapter

Co-authored-by: Max Shuhaliia <mshuhaliia@Maxs-MacBook-Pro-2.local>

mirkorean committed:
Add customParams to yieldlab configuration (#5374)

Customers are using custom parameters to our endpoint to use them downstream. We need a way of supporting this in prebid.

uditalias committed:
Vidazoo Adapter: Feature/user-id (#5386)

* feat(module): multi size request

* fix getUserSyncs
added tests

* update(module): package-lock.json from master

* feat(client): add user id request support

* fix(client): lint errors

Co-authored-by: roman <shmoop207@gmail.com>

jankoulaga committed:
LiveIntentId submodule updates (#5407)

* LiveIntentId submodule.

Bumped the live-connect-js version with changes:
- fixed the problem where userId/index.js was sending the full config on `decode`, while the docs claim it should only send `.params`
- removed the uuid lib
- removed the support for legacy LI first party cookies
- removed the redundant config parameter `providedIdentifierName`
- support for running live-connect inside of multiple wrappers

* LiveIntentId submodule.

Removed reading of non-accessible and therefore redundant code snippets

* Removed excessive config logging.

* Making the publisher id parameter optional.

* Removed unneeded check.

curlyblueeagle committed:
33Across: Adding floors support (#5408)

* check gdpr in buildRequest

* User sync based on whether gdpr applies or not

* check if consent data exists during user sync

* split user sync into further branches: 1) when gdpr does not apply 2) when consent data is unavailable

* contribute viewability to ttxRequest

* update tests

* remove window mock from tests

* use local variables

* introduce ServerRequestBuilder

* add withOptions() method to ServerRequestBuilder

* add semicolons

* sync up package-lock.json with upstream/master

* stub window.top in tests

* introduce getTopWindowSize() for test purpose

* reformat code

* add withSite() method to TtxRequestBuilder

add withSite() method to TtxRequestBuilder

* add isIframe() and _isViewabilityMeasurable()

* handle NON_MEASURABLE viewability in nested iframes

* consider page visibility, stub utils functions getWindowTop() and getWindowSelf()

* contribute viewability as 0 for inactive tab

* add prebidjs version to ttx request

* send caller as an array

* fix JSDoc in utils.js

* send viewability as non measurable when unable to locate target HTMLElement, add warning message

* introduce mapAdSlotPathToElementId()

* introduce getAdSlotHTMLElement(), add logging

* introduce mapAdSlotPathToElementId()

* update logging in ad unit path to element id mapping

* rephrase logging, fix tests

* update adapter documentation

* remove excessive logging

* improve logging

* revert change

* fix return of _mapAdUnitPathToElementId()

* improve logging of _mapAdUnitPathToElementId()

* do not use Array.find()

* return id once element is found

* return id once element is found

* let -> const

* Removing killswitch behavior for GDPR

* Updated comments to reflect current gdpr logic

* URI encode consent string

* Updated example site ID to help Prebid team e2e test our adapter

* send page url in ortb

* Removed redundant pageUrl default

* Restored package-log.json that mirrors prebid's repo

* Sending USP string during buildRequest

* Adding USP consent data to user sync

* add unit test for syncing without bidrequest

* Changed to uspConsent to make the connatation consistent

* Resetting adapter state in adapter after user sync rather than exposing it.

* removed console log

* Adding schain info

* remove setting empty format ext

* better tests invalid values

* removing validation of schain

* Fixed lint errors

* First cut for bidfloors support

* fixed where getFloors is read

* fixed merge conflicts

Co-authored-by: Gleb Glushtsov <gleb.glushtsov@33across.com>
Co-authored-by: Gleb Glushtsov <glebglushtsov@users.noreply.github.com>
Co-authored-by: Gleb Glushtsov <gleb.glushtsov@gmail.com>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-29-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@admins-MacBook-Pro.local>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-137-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@pool-10-1-150-96-nyc.internal.33across.com>
Co-authored-by: Aparna Hegde <ahegde@AHEGDE-MAC.local>
Co-authored-by: Aparna Hegde <ahegde@AHEGDE-MAC.fios-router.home>

uditalias committed:
Vidazoo Adapter: Feature/screen-size (#5385)

* feat(module): multi size request

* fix getUserSyncs
added tests

* update(module): package-lock.json from master

* feat(client): add screen resolution as request param

* feat(client): add `res` param to spec

* fix(client): screen size inside test

Co-authored-by: roman <shmoop207@gmail.com>

justinkuo-appier committed:
Appier: add support for aliases (#5392)

* Add initial partial implementation for Appier bidder adapter.

* Use relative protocol for bidder API url.

* Handle server response for Appier adapter and add related unit tests.

* Support farm-specific prebid server and allow overriding the server with setConfig().

* Add doc for Appier bid adapter.

* Fix const correctness.

* Append requestId to the beacon image URL of Appier adapter to reduce the risks of being cached by proxy servers or browsers.

* Send bidderRequest.refererInfo to Appier bidder server.

* Remove the show beacon since now we generate it in the backend server.

* Only generate a show beacon url if it's not provided by the backend.

* Add version information for Appier adapter using semver (starts from 1.0.0).

* Add a new adapter for Appier bidder.

* Add a new adapter for Appier bidder.

* add appier analyticsAdapter skeleton.

* update initial working version.

* refactor and remove debug messages.

* fix config checking logic.

* implement bidAdjustment and add timeout before send event.

* unify cache operation into cacheManager to avoid direct key/value operation.

* Update server name.

* correct currency impl and message payload. remove unused debug messages.

* update var naming.

* rename creative vars. cleanup comments.

* add test cases for AnalyticsAdapter.

* update test specs file.

* remove spec from main branch.

* add unit tests for AnalyticsAdapter, #1 - Happy cases.

* refactor tests.

* update unit tests for AnalyticsAdapter.

* add tests for bid-adjusted, bid-timeout events

* fix bid adjustment test case.

* add nobid case in analytics unit test.

* add test case for delayed bids and prebidWon messages.

* Use logInfo and logError utilility functions instead of console.log() to print debug messages.

* handle timeout status message correctly.

* correct isTimeout setting logic and test cases.

* replace for...of by array.forEach for IE11

* apply auto formatter.

* refactor: extract timeout logic to a helper function

* tag analytic version with 0.1.0-beta for iCook release

* Re-implement the appier analytics adapter with a simpler design and correctly handle timeouts.

* Remove unused variables.

* Rename methods to improve consistency.

* Code cleanup: rename methods and avoid duplicated code.

* Fix wrong bid response data caused by non-deterministic event ordering of prebid.js.

* Send bid message immediately on auction end without delay.

* add withCredentials in ajax call to get client cookie.

* Fix broken unit test for appier analytic adapter.

* add prediction id support.

* update predictionId/configId format in tests.

* Init refined appier analytics testing

* hotfix: appier analytics support browserstack bug

* feat: add aliases

Co-authored-by: Hong Jen-Yee (PCMan) <pcman.hong@appier.com>
Co-authored-by: Yuan-Hung Huang <yuanhung.huang@appier.com>
Co-authored-by: kdchang <kd.chang@appier.com>
Co-authored-by: chih-ping-weng <chihping.weng@appier.com>

arasaki-yuki committed:
Add GMOSSP Adapter (#5377)

* Add GMOSSP Adapter

* fix test param

uditalias committed:
Vidazoo Adapter: Feature/unit code (#5413)

* feat(module): multi size request

* fix getUserSyncs
added tests

* update(module): package-lock.json from master

* feat(client): send adUnitCode on request payload

Co-authored-by: roman <shmoop207@gmail.com>

jrosendahl committed:
Sovrn - Update Supported ID's, include adunitcode in ad request (#5403)

* added tdid and ad-unit-code

* fixed tdid

* removed digitrust

* repush

* add package-lock from upstream  master

* Delete package-lock.json

* add package-lock from upstream  master

Co-authored-by: Ankit Prakash <aprakash@sovrn.com>
Co-authored-by: Wesley Whitney <wesleywhitney@gmail.com>
Co-authored-by: John Rosendahl <jrosendahl@gmailcom>

coreykress committed:
[Synacormedia] adapter should use format for multi-size banner requests (#5410)

* CAP-1614 - updated docs to show correct size for banner and some other small fixes

* CAP-1636 support schain object in prebid

* CAP-1636 updated the review comments

* CAP-1849 - split up banner and video impressions to use format

Co-authored-by: Corey Kress <corey.kress@synacor.com>
Co-authored-by: Rajkumar Natarajan <rajkumar.natarajan@synacor.com>

jbartek25 committed:
Improve Digital: adapter improvements (#5399)

* Improve Digital: CCPA support

* Outstream video support

* Lint fixes

* Improve Digital: outstream and deal improvements

guiann committed:
Ayl gdp rdefault value (#5391)

* Remove useless bidderCode in bid response

* send all the available sizes in the bid request

* Use the banner sizes if given

* avoid compatibility issue with old bid format

* Remove gdpr default apply value

* minor: use better variable name

* Add unit test on unspecified gdprApplies

Co-authored-by: Guillaume <guiandouard@gmail.com>

robertrmartinez committed:
Price floors new schema support AB Test (#5390)

* Price floors new schema support AB Test

* Add new serve-fast command + lint fix

* update comment

* Only sum up modelWeights once and set as prop!

Fix minor bug in handleFetchResponse to overwrite skipRate

robertrmartinez committed:
Prebid 3.24.0 Release

robertrmartinez committed:
Increment pre version

pm-harshad-mane committed:
Removing Digitrust related test case for PubMatic bidder (#5426)

* added support for pubcommon, digitrust, id5id

* added support for IdentityLink

* changed the source for id5

* added unit test cases

* changed source param for identityLink

* removed digitrust test case

pbrudnick committed:
CCPA modifications in the NextRoll adapter (#5409)

* Add native support

* Add response testing

* DRY test

* Change required from bool to int

* Set mediaType

* Fixes objects

* Fixes object access

* Remove ad property, only set it for banner

* Update tests

* Moving hardcoding values to constants

* Update docs with native information

* Revert "Add native support"

* Getting rid of CCPA adapter validation (#9)

* fix linter errors (#10)

Co-authored-by: Ricardo Azpeitia Pimentel <ricardo.azpeitia@nextroll.com>
Co-authored-by: Abimael Martinez <abijr@users.noreply.github.com>

richaudience committed:
Add Render RichAudience Adapter (#5357)

* Add Render RichAudience Adapter

* Update richaudienceBidAdapter.md & Add Try/Catch

Co-authored-by: sgimenez <sergi.gimenez@richaudience.com>

matthieularere-msq committed:
Mediasquare: Add support for uspConsent + schain userIds support. Plu… (#5396)

* Mediasquare: Add support for uspConsent + schain userIds support. Plus enhance userSync

* fix iframeEnabled and pixelEnabled + suggested shortand statement

smenzer committed:
upgrade id5IdSystem to use v2 of our fetch endpoint (#5406)

- allow publishers to pass deterministic signals
- add a counter to provide analytics on the number of auctions using the id5Id

jaiminpanchal27 committed:
Add tradedesk user id to appnexus adapter (#5346)

* Add tradedesk id support

* Updating appnexus payload for criteo

Co-authored-by: Jaimin Panchal <email@example.com>

invibes committed:
Add TCF2 Support for Invibes (#5378)

* added tcf 2.0

* Updated adapter to support gdprEnforcement

* reverted storage manager initialization

Co-authored-by: florin_nedelcu_invibes <florin.nedelcu@invibes.com>

nickjacob committed:
add AMX adapter (#5383)

hendrikiseke1979 committed:
remove onBidWon callback from adapter (#5414)

* orbidder adapter: add withCredentials:true header to BidRequest and onBidWon Requests

* add blank in order to trigger build again

* remove blank to trigger build ... again

* adding extra line to trigger build ... again

* add prebid version to request

* add unit test for version parameter

* add version parameter to win requests

* fix comment

* trigger rebuild

* trigger rebuild

* remove onBidWon callback from adapter

Co-authored-by: Volk, Rainer <Rainer.Volk@otto.de>
Co-authored-by: RainerVolk4014 <53347752+RainerVolk4014@users.noreply.github.com>
Co-authored-by: siggi-otto <57615762+siggi-otto@users.noreply.github.com>
Co-authored-by: Hendrik Iseke <39734979+hiseke@users.noreply.github.com>
Co-authored-by: Hendrik Iseke <hendrik.iseke@gmx.de>
Co-authored-by: rvolk <>

patmmccann committed:
Make default s2s ttl configurable (#5419)

* make default s2s ttl configurable

AaronColbyPrice committed:
Conversant: update prebid url (#5441)

* Updating Conversant bid adapter URL to new 'cvx'

* Updating Conversant bid adapter URL to new 'cvx' - updating tests to match

* Updating Conversant bid adapter URL to new 'cvx':
rolling back package-lock.json to avoid conflict

patmmccann committed:
Update padsquad for meta.advertiserDomains (#5439)

* Update padsquadBidAdapter_spec.js

* Update padsquadBidAdapter.js

* Update padsquadBidAdapter.js

mamatic committed:
ATS-identityLinkId - add additional info logging events (#5442)

mamatic committed:
ATS-change logError to logInfo type (#5443)

patmmccann committed:
Revert "add AMX adapter (#5383)" (#5455)

This reverts commit d8e5796827a46455185292e4a498628ecdb09bc6.

cciocov committed:
Inskin Bid adapter small changes (#5373)

* Add plr_AdSlot parameter needed by Inskin Pagescroll ad format

* Send additional TCF related information to Inskin's ad server

* Fixed linting issues.

* Added unit tests

uditalias committed:
Vidazoo Adapter: Feature/subdomain (#5446)

* feat(module): multi size request

* fix getUserSyncs
added tests

* update(module): package-lock.json from master

* feat(client): optional server subdomain

* refactor(client): remove unused LOCs

* fix(client): lint issues

Co-authored-by: roman <shmoop207@gmail.com>

vincentproxistore committed:
proxistore bid adapter: delay request to server by 5 min if there were no bids (#5379)

* delay request to server by 5 min if we no ads

* fix testing issue

* use storeManager.js

* change var to const

* add unit test

* remove line and check if user authorizes use of local storage

braizhas committed:
adform and adformOpenRTB bid adapters: Added support for userId modules (#5425)

* adformBidAdapter - added userId module support

* adformOpenRTBBidAdapter - added userId module support

uditalias committed:
Vidazoo Adapter: Feature/bidder-version (#5384)

* feat(module): multi size request

* fix getUserSyncs
added tests

* update(module): package-lock.json from master

* feat(client): add bidder and prebid versions to request params

* fix(client): lint quotes error

Co-authored-by: roman <shmoop207@gmail.com>

onetag-dev committed:
onetagBidAdapter: outstream support (#5435)

* Added full outstream support with onetag renderer, updated request parameters

* Code updates

* Fixies outstream

* Removed wrong pixel calls, code refactoring

* Corregge funzione di calcolo coordinate spazio

* Migliora leggibilità funzione estrazione info pagina

* Fix space coordinates

* Adds bidRequest data

* Updated function interpretResponse

* Minor bug fixing

* Fixes body.scrollHeight call

* Removes Object.assign occurrence

* Updates test file

* Updated payload in request

* Fixes user sync query string

* Sets mediaType into meta field within bidResponse object

* Update documentation in md file

Co-authored-by: valentino <v.v@v.v>
Co-authored-by: Nicola <ciao@ciao>
Co-authored-by: francesco <f.orazini@onetag.com>
